identity-iota/src/credential/credential_validator.rs

```diff
@@ -0,0 +1,645 @@
+      .ok_or(ValidationError::IssuanceDate)
+  }
+
+  // helper method for verifying the signature of a trusted issuer
+  fn extract_corresponding_issuer_helper<'a, T, I: Iterator<Item = &'a ResolvedIotaDocument>>(
+    credential: &Credential<T>,
+    mut issuer_docs: I,
+  ) -> std::result::Result<&'a ResolvedIotaDocument, ValidationError> {
+    let issuer_did: Result<IotaDID> = credential.issuer.url().as_str().parse();
+    match issuer_did {
+      Ok(did) => {
+        // if the issuer_did corresponds to one of the trusted issuers we use the corresponding DID Document to verify
+        // the signature
+        issuer_docs
+          .find(|issuer_doc| issuer_doc.document.id() == &did)
+          .ok_or(ValidationError::IncompatibleIssuerDocuments)
+      }
+      Err(error) => {
+        // the issuer's url could not be parsed to a valid IotaDID
+        Err(ValidationError::IssuerUrl { source: error.into() })
+      }
+    }
+  }
+  /// Verify the signature using the DID Document of a trusted issuer.
+  ///
+  /// This method will only attempt to verify the credential's signature if
+  /// the credential issuer url can be parsed to a DID belonging to one of the trusted issuers.
+  ///
+  /// # Terminology
+  /// This method is a *validation unit*
+  pub fn verify_signature<T: Serialize>(
+    credential: &Credential<T>,
+    trusted_issuers: &[ResolvedIotaDocument],
+    options: &VerifierOptions,
+  ) -> Result<()> {
+    Self::verify_signature_local_error(credential, trusted_issuers, options).map_err(Error::UnsuccessfulValidationUnit)
+  }
+
+  fn verify_signature_local_error<T: Serialize>(
+    credential: &Credential<T>,
+    trusted_issuers: &[ResolvedIotaDocument],
+    options: &VerifierOptions,
+  ) -> ValidationUnitResult {
+    Self::extract_corresponding_issuer_helper(credential, trusted_issuers.iter()).and_then(|issuer| {
+      issuer
+        .document
+        .verify_data(credential, options)
+        .map_err(|error| ValidationError::IssuerProof { source: error.into() })
+    })
+  }
+
+  /// Validates a [Credential].
+  ///
+  /// Common concerns are checked such as the credential's signature, expiration date, issuance date and semantic
+  /// structure.
+  ///
+  ///
+  /// # Errors
+  /// Fails if any of the following conditions occur
+  /// - The structure of the credential is not semantically valid
+  /// - The expiration date does not meet the requirement set in `options`
+  /// - The issuance date does not meet the requirement set in `options`
+  /// - The `issuer` parameter does not provide a DID corresponding to the URL of the credential's issuer.
+  /// - The credential's signature cannot be verified using the issuer's DID Document
+  ///
+  /// Fails on the first encountered error if `fail_fast` is true, otherwise all
+  /// errors will be accumulated in the returned error.
+  // Takes &self in case this method will need some pre-computed state in the future.
+  pub fn full_validation<T: Serialize>(
```

Comment:
Nit: I think just `validate` would be a sufficient name and keep in the convention of using verbs. See previous naming scheme comment.